CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+cmake_minimum_required(VERSION 2.8)
+project( ThetaCloudTests )
+set (CMAKE_CXX_STANDARD 11)
+set(thetaCloudBaseSrc ThetaCloud.cpp)
+set(thetaCloudBaseMockSrc test/fakeIncludes/MockThetaCloud.cpp)
+include_directories(${CMAKE_CURRENT_LIST_DIR})
+
+enable_testing()
+find_package(Threads REQUIRED)
+include(ExternalProject)
+ExternalProject_Add(
+    gtest
+    URL https://github.com/google/googletest/archive/master.zip
+    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/gtest
+    # Disable install step
+    INSTALL_COMMAND ""
+)
+ExternalProject_Get_Property(gtest source_dir binary_dir)
+
+add_library(libgtest IMPORTED STATIC GLOBAL)
+add_dependencies(libgtest gtest)
+set_target_properties(libgtest PROPERTIES
+    "IMPORTED_LOCATION" "${binary_dir}/googlemock/gtest/libgtest.a"
+    "IMPORTED_LINK_INTERFACE_LIBRARIES" "${CMAKE_THREAD_LIBS_INIT}"
+)
+
+add_library(libgmock IMPORTED STATIC GLOBAL)
+add_dependencies(libgmock gtest)
+set_target_properties(libgmock PROPERTIES
+    "IMPORTED_LOCATION" "${binary_dir}/googlemock/libgmock.a"
+    "IMPORTED_LINK_INTERFACE_LIBRARIES" "${CMAKE_THREAD_LIBS_INIT}"
+)
+
+include_directories("${source_dir}/googletest/include"
+                    "${source_dir}/googlemock/include")
+
+set(CMAKE_CXX_FLAGS "-g -O0 -Wall -Werror -fprofile-arcs -ftest-coverage")
+set(CMAKE_C_FLAGS "-g -O0 -Wall -W -fprofile-arcs -ftest-coverage")
+set(CMAKE_EXE_LINKER_FLAGS "-fprofile-arcs -ftest-coverage")
+
+include_directories( test/fakeIncludes )
+set ( fakeCpps test/fakeIncludes/StaticStuff.cpp test/fakeIncludes/Arduino.cpp test/fakeIncludes/FakeUtils.cpp )
+
+set (theta_cloud_base_tests test/main.cpp test/theta_cloud_base_tests.cpp)
+add_executable(thetaCloudBaseTests ${theta_cloud_base_tests} ${thetaCloudBaseSrc} ${fakeCpps})
+target_link_libraries( thetaCloudBaseTests libgtest libgmock gcov )
+add_test(NAME thetaCloudBase COMMAND thetaCloudBaseTests )
+
+set (i2c_helpers_tests test/main.cpp test/i2c_helpers_tests.cpp)
+add_executable(i2cHelpersTests ${i2c_helpers_tests} ${thetaCloudBaseSrc} ${fakeCpps})
+target_link_libraries( i2cHelpersTests libgtest libgmock gcov )
+add_test(NAME i2cHelpers COMMAND i2cHelpersTests )
+
+set (light_tests test/main.cpp test/light_tests.cpp)
+add_executable(lightTests ${light_tests} ThetaCloudLight.cpp ${fakeCpps} ${thetaCloudBaseMockSrc})
+target_link_libraries( lightTests libgtest libgmock gcov )
+add_test(NAME light COMMAND lightTests )
+
+set (humidity_tests test/main.cpp test/humidity_tests.cpp)
```

Comment:
Do you need separate binaries for all of test cases?

Reply:
Not really. Is having tests in separate libraries bad?

The only reason I had was that each of these modules should be independent of each other, this way we can be sure that they can work without other modules